pkg/http/sender.go

```diff
@@ -0,0 +1,638 @@
+				opts = append(opts, WithBasicAuth(user, pass))
+			}
+		case "token":
+			token = v
+			opts = append(opts, WithBearerToken(token))
+		case "auto_flush":
+			if v == "on" {
+				return nil, conf.NewConfigStrParseError("auto_flush option is not supported")
+			}
+		case "auto_flush_rows", "auto_flush_bytes":
+			return nil, conf.NewConfigStrParseError("auto_flush option is not supported")
+		case "min_throughput", "init_buf_size":
+			parsedVal, err := strconv.Atoi(v)
+			if err != nil {
+				return nil, conf.NewConfigStrParseError("invalid %s value, %q is not a valid int", k, v)
+
+			}
+			switch k {
+			case "min_throughput":
+				opts = append(opts, WithMinThroughput(parsedVal))
+			case "init_buf_size":
+				opts = append(opts, WithInitBufferSize(parsedVal))
+			default:
+				panic("add a case for " + k)
+			}
+
+		case "request_timeout", "retry_timeout":
+			timeout, err := strconv.Atoi(v)
+			if err != nil {
+				return nil, conf.NewConfigStrParseError("invalid %s value, %q is not a valid int", k, v)
+			}
+
+			timeoutDur := time.Duration(timeout * int(time.Millisecond))
+
+			switch k {
+			case "request_timeout":
+				opts = append(opts, WithRequestTimeout(timeoutDur))
+			case "retry_timeout":
+				opts = append(opts, WithRetryTimeout(timeoutDur))
+			default:
+				panic("add a case for " + k)
+			}
+		case "tls_verify":
+			switch v {
+			case "on":
+				opts = append(opts, WithTls())
+			case "unsafe_off":
+				opts = append(opts, WithTlsInsecureSkipVerify())
+			default:
+				return nil, conf.NewConfigStrParseError("invalid tls_verify value, %q is not 'on' or 'unsafe_off", v)
+			}
+		case "tls_roots":
+			return nil, conf.NewConfigStrParseError("tls_roots is not available in the go client")
+		case "tls_roots_password":
+			return nil, conf.NewConfigStrParseError("tls_roots_password is not available in the go client")
+		default:
+			return nil, conf.NewConfigStrParseError("unsupported option %q", k)
+		}
+
+	}
+	return NewLineSender(opts...)
+}
+
+// Flush flushes the accumulated messages to the underlying HTTP
+// client. Should be called periodically to make sure that
+// all messages are sent to the server.
+//
+// For optimal performance, this method should not be called after
+// each ILP message. Instead, the messages should be written in
+// batches followed by a Flush call. Optimal batch size may vary
+// from 100 to 1,000 messages depending on the message size and
+// configured buffer capacity.
+func (s *LineSender) Flush(ctx context.Context) error {
+	var (
+		req           *http.Request
+		retryInterval time.Duration
+
+		maxRetryInterval = time.Second
+	)
+
+	err := s.LastErr()
+	s.ClearLastErr()
+	if err != nil {
+		s.DiscardPendingMsg()
+		return err
+	}
+	if s.HasTable() {
+		s.DiscardPendingMsg()
+		return errors.New("pending ILP message must be finalized with At or AtNow before calling Flush")
+	}
+
+	uri := "http"
+	if s.tlsMode > 0 {
+		uri += "s"
+	}
+	uri += fmt.Sprintf("://%s/write", s.address)
```

Comment:
nit: uri doesn't change during the sender lifecycle, so we could allocate it a single time, on sender initialization.